File: JCleanArchitectureApp/appdagger/src/main/java/com/woowrale/jcleanarchitecture/data/local/database/ContactDao.java
package com.woowrale.jcleanarchitecture.data.local.database;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.woowrale.jcleanarchitecture.data.local.model.ContactEntity;

import java.util.List;

@Dao
public interface ContactDao {

    @Query("SELECT * FROM contacts")
    public List<ContactEntity> getAll();

    @Query("SELECT * FROM contacts WHERE id = :id")
    public ContactEntity findById(Integer id);

    @Query("SELECT COUNT(id) FROM contacts")
    public Integer contactCount();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    public void insertContact(List<ContactEntity> contacts);

    @Update
    public void updateContact(ContactEntity contact);
}

File: KCleanArchitectureApp/appdagger/src/main/java/com/woowrale/kcleanarchitecture/data/local/database/ContactDao.kt
package com.woowrale.kcleanarchitecture.data.local.database

import androidx.room.*
import com.woowrale.kcleanarchitecture.data.local.model.ContactEntity

@Dao
interface ContactDao {
    @Query("SELECT * FROM contacts")
    fun getAll(): List<ContactEntity>

    @Query("SELECT * FROM contacts WHERE id = :id")
    fun findById(id: Int): ContactEntity

    @Query("SELECT COUNT(id) FROM contacts")
    fun contactCount(): Int

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertContact(contacts: List<ContactEntity>)

    @Update
    fun updateContact(contact: ContactEntity)
}